kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs

```diff
@@ -18,23 +18,54 @@ impl<'tcx> GotocCtx<'tcx> {
         let label: String = self.current_fn().find_label(&bb);
         // the first statement should be labelled. if there is no statements, then the
         // terminator should be labelled.
+        let check_coverage = self.queries.check_coverage;
         match bbd.statements.len() {
             0 => {
                 let term = bbd.terminator();
-                let tcode = self.codegen_terminator(term);
-                self.current_fn_mut().push_onto_block(tcode.with_label(label));
+                if check_coverage {
+                    let span = term.source_info.span;
+                    let cover = self.codegen_coverage(span);
+                    self.current_fn_mut().push_onto_block(cover.with_label(label));
+                    let tcode = self.codegen_terminator(term);
+                    self.current_fn_mut().push_onto_block(tcode);
+                } else {
+                    let tcode = self.codegen_terminator(term);
+                    self.current_fn_mut().push_onto_block(tcode.with_label(label));
+                }
             }
             _ => {
                 let stmt = &bbd.statements[0];
-                let scode = self.codegen_statement(stmt);
-                self.current_fn_mut().push_onto_block(scode.with_label(label));
+                if check_coverage {
+                    let span = stmt.source_info.span;
+                    let cover = self.codegen_coverage(span);
+                    self.current_fn_mut().push_onto_block(cover.with_label(label));
+                    let scode = self.codegen_statement(stmt);
+                    self.current_fn_mut().push_onto_block(scode);
+                } else {
+                    let scode = self.codegen_statement(stmt);
+                    self.current_fn_mut().push_onto_block(scode.with_label(label));
+                }
 
                 for s in &bbd.statements[1..] {
+                    if check_coverage {
+                        let span = s.source_info.span;
+                        // TODO: Push cover statement based on some TBD condition
+                        let cover = self.codegen_coverage(span);
+                        self.current_fn_mut().push_onto_block(cover);
+                    }
                     let stmt = self.codegen_statement(s);
                     self.current_fn_mut().push_onto_block(stmt);
                 }
-                let term = self.codegen_terminator(bbd.terminator());
-                self.current_fn_mut().push_onto_block(term);
+                // TODO: test with division by zero or overflow that false
```

Comment:
please address this and remove the TODO

Reply:
BTW, I see `div-zero` tests.So I'm assuming you can remove this todo already.

Reply:
Included in the latest changes.